File: 30 days of code/Day 04 - Class vs. instance/Class vs. instance.cs
/**
 *              CLASS VS. INSTANCE | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 04
 */

using System;
using System.IO;

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        int testCases = int.Parse(read.ReadLine());
        while (testCases-- > 0)
        {
            int age = int.Parse(read.ReadLine());
            Person person = new Person(age);
            person.AmIOld();
            for(int i = 0; i < 3; ++i)
            {
                person.YearPasses();
            }
            person.AmIOld();
            Console.WriteLine();
        }
        read.Close();
    }
}

class Person
{
    private int Age { get; set; }

    public Person(int initialAge)
    {
        if (initialAge < 0)
        {
            Console.WriteLine("Age is not valid, setting age to 0.");
            Age = 0;
        }
        else
        {
            Age = initialAge;
        }
    }

    public void AmIOld()
    {
        if (Age < 13)
        {
            Console.WriteLine("You are young.");
        }
        else if (Age < 18)
        {
            Console.WriteLine("You are a teenager.");
        }
        else
        {
            Console.WriteLine("You are old.");
        }
    }

    public void YearPasses()
    {
        Age++;
    }
}

File: 30 days of code/Day 04 - Class vs. instance/Class vs. instance.cpp
/**
*              CLASS VS. INSTANCE | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 04
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>

using namespace std;

ifstream file("input.txt");
istream& f = file;

class Person {
public:
	Person(int initlialAge) {
		if (initlialAge < 0) {
			cout << "Age is not valid, setting age to 0." << endl;
			this->age = 0;
		} else {
			this->age = initlialAge;
		}
	}

	void amIOld() {
		if (age < 13) {
			cout << "You are young." << endl;
		} else if (age < 18) {
			cout << "You are a teenager." << endl;
		} else {
			cout << "You are old." << endl;
		}
	}

	void yearPasses() {
		this->age++;
	}
private:
	int age;
};

void main() {
	int testCases; f >> testCases;
	while (testCases-- > 0) {
		int age; f >> age;
		Person person(age);
		person.amIOld();
		for (int i = 0; i < 3; ++i) {
			person.yearPasses();
		}
		person.amIOld();
		cout << '\n';
	}
}